src/librustc_mir/hair/cx/expr.rs

```diff
@@ -718,7 +718,23 @@ fn make_mirror_unadjusted<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
                 ExprKind::Cast { source }
             }
         }
-        hir::ExprKind::Type(ref source, _) => return source.make_mirror(cx),
+        hir::ExprKind::Type(ref source, ref ty) => {
+            if let Some(user_ty) = cx.tables.user_provided_tys().get(ty.hir_id) {
```

Comment:
I think I should use `expect` here instead of silently returning the mirror of the source expression when we can't find the user_ty in the table.